Makefile.build

```diff
@@ -1,389 +0,0 @@
-endif
-endif
-pretty-timed:
-	$(HIDE)$(MAKE) --no-print-directory make-pretty-timed
-	$(HIDE)$(MAKE) --no-print-directory print-pretty-timed
-.PHONY: pretty-timed make-pretty-timed make-pretty-timed-before make-pretty-timed-after print-pretty-timed print-pretty-timed-diff
-
-ifneq (,$(TIMING))
-TIMING_ARG=-time
-ifeq (after,$(TIMING))
-TIMING_EXT=after-timing
-else
-ifeq (before,$(TIMING))
-TIMING_EXT=before-timing
-else
-TIMING_EXT=timing
-endif
-endif
-else
-TIMING_ARG=
-endif
-
-###########################################################################
-# Build the .v deps
-VFILED=.vfiles.d
--include $(VFILED)
-
-# All dependency includes must be declared secondary, otherwise make will
-# delete them if it decided to build them by dependency instead of because
-# of include, and they will then be automatically deleted, leading to an
-# infinite loop.
-
-.SECONDARY: $(VFILED)
-
-###########################################################################
-# Compilation options
-###########################################################################
-
-# Default timing command
-# Use /usr/bin/env time on linux, gtime on Mac OS
-TIMEFMT?="$@ (real: %e, user: %U, sys: %S, mem: %M ko)"
-ifneq (,$(TIMED))
-ifeq (0,$(shell /usr/bin/env time -f "" true >/dev/null 2>/dev/null; echo $$?))
-STDTIME?=/usr/bin/env time -f $(TIMEFMT)
-else
-ifeq (0,$(shell gtime -f "" true >/dev/null 2>/dev/null; echo $$?))
-STDTIME?=gtime -f $(TIMEFMT)
-else
-STDTIME?=time
-endif
-endif
-else
-STDTIME?=/usr/bin/env time -f $(TIMEFMT)
-endif
-
-TIMER=$(if $(TIMED), $(STDTIME), $(TIMECMD))
-
-# NB: do not use a variable named TIME, since this variable controls
-# the output format of the unix command time. For instance:
-#   TIME="%C (%U user, %S sys, %e total, %M maxres)"
-
-PLUGININCLUDES=$(addprefix -I _build/default/, $(wildcard plugins/*))
-DUNEPLUGININCLUDES=$(addprefix -I _build/default/, $(wildcard plugins/*))
-NATIVEINCLUDES=$(addprefix -nI _build/default/, kernel/.kernel.objs/byte)
-
-ifdef NATIVECOMPUTE
-  COQOPTS += -w -deprecated-native-compiler-option -native-compiler ondemand
-endif
-COQOPTS += $(COQWARNERROR) $(COQUSERFLAGS)
-# Beware this depends on the makefile being in a particular dir, we
-# should pass an absolute path here but windows is tricky
-# c.f. https://github.com/coq/coq/pull/9560
-BOOTCOQC=$(TIMER) $(COQC) -coqlib $(VO_OUT_DIR) -q $(COQOPTS) $(DUNEPLUGININCLUDES)
-
-###########################################################################
-# Infrastructure for the rest of the Makefile
-###########################################################################
-
-VO_TOOLS_DEP = $(COQC)
-
-ifeq ($(BEST),byte)
-  VO_TOOLS_DEP += $(CONTEXT)/lib/stublibs/dllcoqrun_stubs.so
-endif
-
-ifdef NATIVECOMPUTE
-  VO_TOOLS_DEP += $(COQNATIVE)
-endif
-
-ifdef VALIDATE
- VO_TOOLS_DEP += $(CHICKEN)
-endif
```

Comment:
Validate job?

Reply:
The makefile used to validate by accumulating all the vo files. I have a dune stanza that is able to do that (but for dune 3.0 or before but a bit more painful).

The other solution is to do a new incremental validation (producing a chk log or something). There are flags we can pass to coqchk to allow for this, and worries about the coqchk binary or vo files being swapped out shouldn't hold since the rules will depend on those anyway.

I would suggest the first, and then we can work on the second at a later date.

Reply:
There are indeed several ways to do validate, but for this PR, is the way we are doing it bad for you?

Reply:
AFAICT we have no validate job anymore.

Reply:
Yes we do, unless I'm looking to the wrong CI

Reply:
I think there is some confusion. There are two "validate" jobs. One is far calling coqchk on vo files, the other is turning a vio to a vok file. The CI is only doing the latter, the coqchk target has been removed.

Reply:
No you are right, it is there in the CI. The makefile target is still missing however.

Reply:
CI doesn't use the makefiles to call coqchk
see https://gitlab.com/coq/coq/-/jobs/2506489738#L427

Reply:
Yes, I see that now. I guess removing this is OK then. I'll create an issue for adding a build rule for coqchk in the future.

Reply:
IMO we should still provide a validate target, and since it would operate on _build/install we could make CI use it too.

Reply:
@SkySkimmer I've created https://github.com/coq/coq/issues/16072 and I am fairly certain we can replicate both kinds in dune.